homeassistant/components/weatherflow/event.py

```diff
@@ -0,0 +1,145 @@
+            format_dispatch_call(config_entry),
+            async_add_events,
+        )
+    )
+
+
+class _WeatherFlowEventEntity(EventEntity):
+    """Base WeatherFlow event entity."""
+
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+
+    def __init__(
+        self,
+        device: WeatherFlowDevice,
+        description: EventEntityDescription,
+    ) -> None:
+        """Initialize the base WeatherFlow event entity."""
+
+        self.device = device
+        self.entity_description = description
+
+        translation_key = description.translation_key
+        self._attr_translation_key = (
+            translation_key if isinstance(translation_key, str) else None
+        )
+        name = description.name
+        self._attr_name = name if isinstance(name, str) else None
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, device.serial_number)},
+            manufacturer="WeatherFlow",
+            model=device.model,
+            name=device.serial_number,
+            sw_version=device.firmware_revision,
+        )
+        self._attr_unique_id = f"{device.serial_number}_{description.key}"
+        self._attr_entity_id = (
+            f"event.weatherflow_{device.serial_number}_{description.key}"
+        )
+
+
+class WeatherFlowRainStartEventEntity(_WeatherFlowEventEntity):
+    """Event entity that fires when rain starts."""
+
+    entity_description = RAIN_START_DESCRIPTION
+    _attr_icon = "mdi:weather-rainy"
+
+    def __init__(self, device: WeatherFlowDevice) -> None:
+        """Initialize a rain start event entity."""
+
+        super().__init__(device, RAIN_START_DESCRIPTION)
+
+    async def async_added_to_hass(self) -> None:
+        """Subscribe to rain start events."""
+        self.async_on_remove(self.device.on(EVENT_RAIN_START, self._handle_event))
+
+    @callback
+    def _handle_event(self, event) -> None:
+        self._trigger_event(
+            "rain_start",
+            {
+                "timestamp": event.timestamp.isoformat()
+                if getattr(event, "timestamp", None)
+                else None
+            },
+        )
+        self.async_write_ha_state()
+
+
+class WeatherFlowLightningStrikeEventEntity(_WeatherFlowEventEntity):
+    """Event entity that fires when lightning strikes."""
+
+    entity_description = LIGHTNING_STRIKE_DESCRIPTION
+    _attr_icon = "mdi:weather-lightning"
+
+    def __init__(self, device: WeatherFlowDevice) -> None:
+        """Initialize a lightning strike event entity."""
+
+        super().__init__(device, LIGHTNING_STRIKE_DESCRIPTION)
+
+    async def async_added_to_hass(self) -> None:
+        """Subscribe to lightning strike events."""
+        self.async_on_remove(self.device.on(EVENT_STRIKE, self._handle_event))
+
+    @callback
+    def _handle_event(self, event) -> None:
+        self._trigger_event(
+            "strike",
+            {
+                "timestamp": event.timestamp.isoformat()
```

Comment:
Why do we have this timestamp here? Isn't this already the moment we got this event?

Reply:
your right, was a left over, wanted somethint to do with it, it also included energy and distance. aready removed that. pushed a new commit